src/plotting/plotWindowPortal.component.tsx

```diff
@@ -57,6 +45,141 @@ class PlotWindowPortal extends React.PureComponent<
+      chartjsZoomScript.src =
+        'https://cdnjs.cloudflare.com/ajax/libs/chartjs-plugin-zoom/1.2.1/chartjs-plugin-zoom.js';
+      chartjsZoomScript.crossOrigin = 'anonymous';
+      chartjsZoomScript.referrerPolicy = 'no-referrer';
+      chartjsZoomScript.async = false;
+      chartjsZoomScript.defer = false;
+      externalWindow.document.head.appendChild(chartjsZoomScript);
+
+      const chartjsDateFnsScript = document.createElement('script');
+      // TODO: switch this to cdnjs once it's added - this is for consistency and so we can use renovate to update it
+      chartjsDateFnsScript.src =
+        'https://cdn.jsdelivr.net/npm/chartjs-adapter-date-fns@2.0.0/dist/chartjs-adapter-date-fns.bundle.min.js';
+      chartjsDateFnsScript.crossOrigin = 'anonymous';
+      chartjsDateFnsScript.referrerPolicy = 'no-referrer';
+      chartjsDateFnsScript.async = false;
+      chartjsDateFnsScript.defer = false;
+      externalWindow.document.head.appendChild(chartjsDateFnsScript);
+
+      const chartjsCode = document.createElement('script');
+      chartjsCode.type = 'text/javascript';
+
+      /**
+       * This code in the below string (which gets inserted into the script tag)
+       * does the following:
+       * `waitForElm` - given a selector, returns a promise that resolves with the element
+       * using a `MutationObserver` to inspect DOM changes - used to wait for Chart.js canvas element to be loaded by React
+       * `waitForChartJS` - is a simple `setInterval` that checks if the chart.js object has loaded before running any Chart.js code
+       * `MutationObserver` code - we need a way to pass the `data` and `options` variables from
+       * React in the main window to the Chart.js code. We do this by using data-* attributes on the canvas element,
+       * which React can set (see plot.component.tsx). The MutationObserver thus watches for changes to the canvas object,
+       * which then updates Chart.js if necessary
+       * In the mutation observer code, when options are updated we set the legend filter function to filter out datasets
+       * which have been set to transparent (which is done via the show/hide buttons)
+       */
+      const code = `
+      function waitForElm(selector) {
+        return new Promise(resolve => {
+          if (document.querySelector(selector)) {
+            return resolve(document.querySelector(selector));
+          }
+  
+          const observer = new MutationObserver(mutations => {
+            if (document.querySelector(selector)) {
+              resolve(document.querySelector(selector));
+              observer.disconnect();
+            }
+          });
+  
+          observer.observe(document.body, {
+            childList: true,
+            subtree: true
+          });
+        });
+      }
+
+      var waitForChartJS = setInterval(function () {
+        if (typeof Chart !== 'undefined' && typeof Hammer !== 'undefined' && typeof ChartZoom !== 'undefined') {
```

Comment:
Do we also need to wait for _date-fns_? Or is this not necessary because we won't be using it to format dates anymore, since this is covered in `displayFormats`?

Reply:
We don't actually import all of `date-fns` - the import we do for `chartjs-adapter-date-fns` includes any date-fns code it uses. I did try and find the variable so we can wait for it, but I don't think it creates one. Either way, the browser should wait for it to load as I set `async` & `defer` to be false on all the libraries being loaded - these checks were from when I didn't realise some browsers set `async = true` if you dynamically add a script but I figured they act as another safety net so it's good to keep the checks in.

I'll double check if there's a variable we can wait for with the date library tomorrow